ml-algorithms/src/main/java/org/opensearch/ml/engine/encryptor/EncryptorImpl.java

```diff
@@ -247,16 +251,46 @@ private void initializeNewMasterKey(
     }
 
     private PutDataObjectRequest createPutDataObjectRequest(String tenantId, String generatedMasterKey) {
+        String masterKeyId = MASTER_KEY;
+        if (tenantId != null) {
+            masterKeyId = MASTER_KEY + "_" + hashString(tenantId);
+        }
         return PutDataObjectRequest
             .builder()
             .tenantId(tenantId)
             .index(ML_CONFIG_INDEX)
-            .id(MASTER_KEY)
+            .id(masterKeyId)
             .overwriteIfExists(false)
-            .dataObject(ImmutableMap.of(MASTER_KEY, generatedMasterKey, CREATE_TIME_FIELD, Instant.now().toEpochMilli()))
+            .dataObject(
+                Map
+                    .of(
+                        MASTER_KEY,
+                        generatedMasterKey,
+                        CREATE_TIME_FIELD,
+                        Instant.now().toEpochMilli(),
+                        TENANT_ID,
+                        Objects.requireNonNullElse(tenantId, DEFAULT_TENANT_ID)
+                    )
+            )
             .build();
     }
 
+    private String hashString(String input) {
+        try {
+            // Create a MessageDigest instance for SHA-256
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
+
+            // Perform the hashing and get the byte array
+            byte[] hashBytes = digest.digest(input.getBytes());
+
+            // Convert the byte array to a Base64 encoded string
+            return Base64.getEncoder().encodeToString(hashBytes);
```

Comment:
This creates slash characters in the encoding, which would break requests to a remote cluster going over the REST path.